### PR TITLE
test: refactor inspect-test

### PR DESCRIPTION
**Pull request by joyeecheung**
- Make the test messages more informative.
- Refactor the test flow so that we could skip flaky properties
  (currently the potentially neutered typed array)
  and that one missing property does not stop later tests.

Refs: https://github.com/nodejs/llnode/issues/150
We should now be able to skip https://github.com/nodejs/llnode/issues/150 , it remains a question whether this should be fixed (or is it fixable?)

<details>
<summary>TAP output of inspect-test.js before this patch</summary>

```
TAP version 13
# v8 inspect
ok 1 undefined
ok 2 null
ok 3 method should have `this`
ok 4 null
ok 5 addr of `Class` should match
ok 6 null
ok 7 .x smi property
ok 8 .y heap number property
ok 9 .hashmap object property
ok 10 null
ok 11 addr of `Object` should match
ok 12 null
ok 13 [0] null element
ok 14 [4] undefined element
ok 15 [23] RegExp element
ok 16 [25] Arrow Function element
ok 17 .some-key property
ok 18 .other-key property
ok 19 .array JSArray property
ok 20 .array JSArray property
ok 21 .array-buffer JSArrayBuffer property
ok 22 .uint8-array JSArrayBufferView property
ok 23 .buffer JSArrayBufferView property
ok 24 .cons-string ConsString property
ok 25 .thin-string ThinString property
ok 26 .externalized-string ExternalString property
ok 27 .sliced-externalized-string Sliced ExternalString property
ok 28 null
ok 29 regexp.source
ok 30 null
ok 31 cons string content
ok 32 null
ok 33 --string-length truncates the string
ok 34 null
ok 35 thin string content
ok 36 null
ok 37 array length
ok 38 array content
ok 39 null
ok 40 long array length
ok 41 long array content
ok 42 null
ok 43 array buffer content
ok 44 null
ok 45 array buffer content with maximum length 1
ok 46 null
ok 47 typed array content
ok 48 null
ok 49 typed array content with maximum length 1
ok 50 null
ok 51 buffer content
ok 52 null
ok 53 buffer content with maximum length 1
ok 54 null
ok 55 arrow method source
ok 56 null
ok 57 method source found
ok 58 method.previous
ok 59 method.scopedValue
ok 60 `method` should have `scopedAPI`
ok 61 `method` should have `closure`
ok 62 null
ok 63 method.scopedAPI.internalFields
ok 64 null
ok 65 method.closure.outerVar

1..65
# tests 65
# pass  65

# ok
```
</details>

<details>
<summary>TAP output of inspect-test.js after this patch (removed a few redundant `t.error(err) in callbacks`)</summary>

```TAP version 13
# v8 inspect
ok 1 method should have `this` and `fn`
ok 2 method[[context]].(previous) should exist 
ok 3 method[[context]].(closure) should exist 
ok 4 method[[context]].scopedVar should exist 
ok 5 method[[context]].scopedAPI should exist 
ok 6 method[[context]].scopedArray should exist 
ok 7 method[[context]].closure.outerVar should exist
ok 8 method[[context]].(closure) validator should complete
ok 9 method[[context]].closure.scopedAPI should have internal fields
ok 10 method[[context]].scopedAPI validator should complete
ok 11 method[[context]].scopedArray[0] should be `<Smi: 0>`
ok 12 method[[context]].scopedArray[1] should be a Zlib
ok 13 method[[context]].scopedArray validator should complete
ok 14 v8 inspect -s <method> should return output
ok 15 method source should match
ok 16 address in the detailed view of Class should match the address in the parent's view
ok 17 this.x should be <Smi: 1>
ok 18 this.y should be 123.456
ok 19 this.hashmap should be an object
ok 20 address in the detailed view of Object should match the address in the parent's view
ok 21 hashmap[0] null element should exist 
ok 22 hashmap[4] undefined element should exist 
ok 23 hashmap[23] RegExp element should exist 
ok 24 hashmap[25] Arrow Function element should exist 
ok 25 hashmap.some-key property should exist 
ok 26 hashmap.other-key property should exist 
ok 27 hashmap.cons-string ConsString property should exist 
ok 28 hashmap.internalized-string Internalized String property should exist 
ok 29 hashmap.thin-string ThinString property should exist 
ok 30 hashmap.externalized-string ExternalString property should exist 
ok 31 hashmap.sliced-externalized-string Sliced ExternalString property should exist 
ok 32 hashmap.array JSArray property should exist 
ok 33 hashmap.long-array JSArray property should exist 
ok 34 hashmap.array-buffer JSArrayBuffer property should exist 
ok 35 hashmap.uint8-array JSArrayBufferView property should exist 
ok 36 hashmap.buffer JSArrayBufferView property should exist 
ok 37 hashmap[23] should have the correct regexp.source
ok 38 hashmap[23] RegExp element validator should complete
ok 39 hashmap[25] should have the correct function source
ok 40 hashmap[25] Arrow Function element validator should complete
ok 41 hashmap.cons-string should have the right content
ok 42 hashmap.cons-string ConsString property validator #0 should complete
ok 43 --string-length should truncate the string
ok 44 hashmap.cons-string ConsString property validator #1 should complete
ok 45 hashmap.thin-string should have the right content
ok 46 hashmap.thin-string ThinString property validator should complete
ok 47 length of hashmap.array should be 6
ok 48 hashmap.array JSArray property validator should complete
ok 49 length of hashmap.long-array should be 6
ok 50 hashmap.long-array should have the right content
ok 51 hashmap.long-array JSArray property validator should complete
ok 52 hashmap.array-buffer should have the right content
ok 53 hashmap.array-buffer JSArrayBuffer property validator #0 should complete
ok 54 hashmap.array-buffer should have the right content with --array-length 1
ok 55 hashmap.array-buffer JSArrayBuffer property validator #1 should complete
ok 56 hashmap.uint8-array should have the right content
ok 57 hashmap.uint8-array JSArrayBufferView property validator #0 should complete
ok 58 hashmap.uint8-array should have the right content with --array-length 1
ok 59 hashmap.uint8-array JSArrayBufferView property validator #1 should complete
ok 60 hashmap.uint8-array should have the right content
ok 61 hashmap.buffer JSArrayBufferView property validator #0 should complete
ok 62 hashmap.buffer should have the right content with --array-length 1
ok 63 hashmap.buffer JSArrayBufferView property validator #1 should complete

1..63
# tests 63
# pass  63

# ok
```
</details>

### CODE REVIEW

File: test/inspect-test.js [hunk 1]
```diff
@@ -4,368 +4,554 @@ const tape = require('tape');
...
+  },
+  // .uint8-array=0x0000393071133e59:<ArrayBufferView: backingStore=0x000000000195b230, byteOffset=0, byteLength=6>,
+  // OR
+  // .uint8-array=0x000003df9cbe7eb9:<ArrayBufferView [neutered]>,
+  'uint8-array': {
+    re: new RegExp('.uint8-array=(0x[0-9a-f]+):<ArrayBufferView: ' +
+                    'backingStore=0x[0-9a-f]+, byteOffset=\\d+, ' +
+                    'byteLength=6>'),
+    desc: '.uint8-array JSArrayBufferView property',
+    optional: 'might be neutered',
```
Comment (joyeecheung) [inline test/inspect-test.js]: This is where the test is configured to be skippable (if we could not find such property, the validator won't be run, a skip message with the reason specified in `optional` will be printed)

**Comment by joyeecheung** [comment]
CI now skips the neutered typed array

https://travis-ci.org/nodejs/llnode/jobs/354456004

```
ok 44 .uint8-array JSArrayBufferView property: might be neutered # SKIP
```

File: test/inspect-test.js [hunk 2]
```diff
@@ -4,368 +4,554 @@ const tape = require('tape');
...
+      }
+    }
+  }
+  return addresses;
+}
 
-  sess.linesUntil(/^>/, (err, lines) => {
-    t.error(err);
-    lines = lines.join('\n');
+// To avoid lldb output being mixed, we must do the validation one by one
```
Comment (bnoordhuis) [inline test/inspect-test.js]: What does this mean, mixed?

File: test/inspect-test.js (hunk 1 above)
Comment (bnoordhuis) [reply test/inspect-test.js]: Would it be possible to use an array of regexes and check if one matches?  This `.optional` property means the test can never fail.  That's undesirable, IMO.

File: test/inspect-test.js (hunk 2 above)
Comment (joyeecheung) [reply test/inspect-test.js]: I think if simply we loop over the validators, it is possible that one `sess.linesUntil()` would consume the output of `sess.send()` from another validator (if one of them fails and receives the output from another that accidentally contains the marker that it's waiting for, the test results could be pretty bizzare).

**Comment by joyeecheung** [comment]
Updated the optional regexp per suggestion by @bnoordhuis .

**Pull request merged**